crates/sui-framework/sources/publisher.move

```diff
@@ -7,34 +7,37 @@
 module sui::publisher {
     use sui::object::{Self, UID};
     use sui::tx_context::{TxContext, sender};
-    use std::ascii::{Self, String};
-    use std::type_name;
-    use std::vector;
+    use std::type_name::{Self, TypeName};
+    use std::ascii::String;
     use sui::types;
 
     /// Tried to claim ownership using a type that isn't a one-time witness.
     const ENotOneTimeWitness: u64 = 0;
 
-    /// ASCII character code for the `:` (colon) symbol.
-    const ASCII_COLON: u8 = 58;
-
     /// This type can only be created in the transaction that
     /// generates a module, by consuming its one-time witness, so it
     /// can be used to identify the address that published the package
     /// a type originated from.
     struct Publisher has key, store {
         id: UID,
-        type: String
+        type: TypeName,
```

Comment:
Is there a reason to keep the `type` here?  I only see `package` and `module_name` being used.

Reply:
Good question. It's good for visibility in a sense that people know the full OTW. But on the other hand we could remove it as unused.

Reply:
We can probably skip it, right? The OTW can only be one thing, which you can always derive from the moduel name and address.

Reply:
Addressed your comment (and other ones).